### PR TITLE
Add users

### PR DESCRIPTION
**Pull request by crookedneighbor**

### CODE REVIEW

File: lib/haxfred-irc.js [hunk 1]
```diff
@@ -59,7 +65,27 @@ var haxfred_irc = function(haxfred) {
...
   });
 
+  Haxfred.irc.client.addListener('part', function (channel, nick) {
+    // Remove nick to array of users
+    var removeNick = Haxfred.irc.users[channel].indexOf(nick);
+    if(removeNick > -1) {
+      Haxfred.irc.users[channel].splice(removeNick, 1);
+    }
+
+     haxfred.emit('irc.part',{
```
Comment (lejeunerenard) [inline lib/haxfred-irc.js:75]: Good call on adding a `irc.part` event.

File: lib/haxfred-irc.js [hunk 2]
```diff
@@ -52,6 +53,11 @@ var haxfred_irc = function(haxfred) {
 
   /* ----- Listeners ----- */
   Haxfred.irc.client.addListener('join', function (channel, nick) {
+    // Add nick to array of users
+    if(Haxfred.irc.users[channel]) {
+      Haxfred.irc.users[channel].push(nick);
```
Comment (lejeunerenard) [inline lib/haxfred-irc.js:58]: I think you may have to declare `Haxfred.irc.users[channel]` an array before doing this. You do declare the array in the listener on irc's `names` event, but if `join` is emitted before `names` when Haxfred first joins, you'd get an error trying to push to `Haxfred.irc.users[channel]`. I believe.

Comment (crookedneighbor) [inline lib/haxfred-irc.js:58]: That's why there's the if statement.

Comment (crookedneighbor) [inline lib/haxfred-irc.js:58]: I ran into just that problem. I can add a comment in with the if statement to make that clearer.

Comment (lejeunerenard) [inline lib/haxfred-irc.js:58]: Ah. Fair enough. I guess you will get the `join`ed nick in the `names` event anyways. Good point.

File: lib/haxfred-irc.js [hunk 3]
```diff
@@ -59,7 +65,27 @@ var haxfred_irc = function(haxfred) {
...
+     });
+  });
+
   Haxfred.irc.client.addListener('names', function(channel, nicks) {
+    // Update the room with an array of users
+    Haxfred.irc.users[channel] = [];
+    for(var n in nicks) {
+     Haxfred.irc.users[channel].push(n);
+    }
+
```
Comment (lejeunerenard) [inline lib/haxfred-irc.js:88]: We can leave this for another PR but while we are adding events to emit, we should add a `irc.names` so that components can know when the names have changed if they already used the `Haxfred.irc.users` variable. It will of course emit several times when Haxfred tries to change his name, but that shouldnt be a problem.
